packages/data-context/src/actions/ProjectActions.ts

```diff
@@ -645,4 +658,42 @@ export class ProjectActions {
     debug('navigating to Debug page')
     this.api.routeToDebug()
   }
+
+  /**
+   * Fetches basic information about an associated Cloud project.
+   * Tries to use cached data, and falls back to fetching the data
+   * from the remote endpoint.
+   */
+  async fetchCloudProjectInfo (): Promise<CoreDataShape['cloud'] | undefined> {
+    const projectSlug = await this.ctx.project.projectId()
+
+    if (this.ctx.coreData.cloud.id) {
```

Comment:
Caching - we only need to fetch this once.

Reply:
It does not appear that this caching would ever get cleared.  Would it?  What about running in Global Mode and switching between projects?   Or reconnecting a local project to a different Cypress Cloud project?

Reply:
I thought we clear the entire `data-context` between projects - this may not be the case, let me find out.

Reply:
Changed this, please see the updated code. We still cache, but the data is it a separate key (`cloud`) and we have a single place to clear it between projects based on the existing pattern of calling `lifecycle.clearCurrentProject()`